rfcs/text/semantic-release.md

```diff
@@ -0,0 +1,249 @@
+> 1. footers other than `BREAKING CHANGE: <description> `may be provided and follow a convention similar to [git trailer format](https://git-scm.com/docs/git-interpret-trailers).
+>
+> Additional types are not mandated by the conventional commits specification, and have no implicit effect in semantic versioning (unless they include a BREAKING CHANGE).
+>
+> A scope may be provided to a commit’s type, to provide additional contextual information and is contained within parenthesis, e.g., `feat(parser): add ability to parse arrays`.  
+> [[Source]](https://www.conventionalcommits.org/en/v1.0.0/)
+
+> ## Subject
+> The subject contains a succinct description of the change:
+>
+> * use the imperative, present tense: "change" not "changed" nor "changes"
+> * don't capitalize the first letter
+> * no dot (.) at the end
+> ## Body
+> Just as in the *subject*, use the imperative, present tense: "change" not "changed" nor "changes". The body should include the motivation for the change and contrast this with previous behavior.
+>
+> ## Footer
+> The footer should contain any information about **Breaking Changes** and is also the place to reference GitHub issues that this commit **Closes**.
+>
+> **Breaking Changes** should start with the word `BREAKING CHANGE:` with a space or two newlines. The rest of the commit message is then used for this.  
+> [[Source]](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit)
+
+We will use [`commitlint`](https://github.com/conventional-changelog/commitlint) and [`husky`](https://github.com/typicode/husky) to check our commits against the `conventionalcommits` guidelines.
+- [`husky`](https://github.com/typicode/husky) will manage the git commit hooks defined in `package.json`
+- we will write a hook that runs the commit message via [`commitlint`](https://github.com/conventional-changelog/commitlint)
+- we will use the [`@commitlint/config-conventional`](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional) config
+- [`husky`](https://github.com/typicode/husky) will abort the commit if the commit message is in the wrong format
+
+We will use [`commitizen`](https://github.com/commitizen/cz-cli) to help write commit messages
+- we will use the [`cz-conventional-changelog`](https://github.com/commitizen/cz-conventional-changelog) config
+- users can choose to write commit messages with the regular `git` CLI or in their IDE still
+- users can launch [`commitizen`](https://github.com/commitizen/cz-cli) with `npm run commit`
+
+We will add [`probot/semantic-pull-requests`](https://probot.github.io/apps/semantic-pull-requests/) as a GitHub check.
+This will prevent PR's being merged that do not follow the correct syntax.
+
+## Reverting a Release
+> If you have introduced a breaking bug in a release you have 2 options:
+> * If you have a fix immediately ready, commit and push it (or merge it via a pull request) to the release branch
```

Comment:
release branch as in `release` or does his mean `master`?

Reply:
This means any of the release branches
`['+([0-9])?(.{+([0-9]),x}).x', 'master', 'next', 'next-major', {name: 'beta', prerelease: true}, {name: 'alpha', prerelease: true}]`
https://github.com/semantic-release/semantic-release/blob/916c2685c57f3490fb1e50afbf72ea8dce11e188/docs/usage/configuration.md#branches